src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,264 @@
+    {
+        return customName != null;
+    }
+
+    /**
+     * Get the formatted ChatComponent that will be used for the sender's username in chat
+     */
+    @Override
+    public ITextComponent getDisplayName()
+    {
+        return new TextComponentString(getName());
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT()
+    {
+        final NBTTagCompound compound = new NBTTagCompound();
+
+        int index = 0;
+        final NBTTagList handlerList = new NBTTagList();
+        final NBTTagList indexList = new NBTTagList();
+        for(IItemHandlerModifiable handlerModifiable : handlers) {
+            if (handlerModifiable instanceof INBTSerializable)
+            {
+                INBTSerializable serializable = (INBTSerializable) handlerModifiable;
+                handlerList.appendTag(serializable.serializeNBT());
+                indexList.appendTag(new NBTTagInt(index++));
+            }
+        }
+
+        compound.setTag(NBT_KEY_HANDLERS, handlerList);
+
+        if (hasCustomName()) {
+            compound.setString(NBT_KEY_NAME, customName);
+        }
+
+        return compound;
+    }
+
+    @Override
+    public void deserializeNBT(final NBTTagCompound nbt)
+    {
+        final NBTTagList handlerList = nbt.getTagList(NBT_KEY_NAME, Constants.NBT.TAG_COMPOUND);
+        final NBTTagList indexList = nbt.getTagList(NBT_KEY_HANDLERS_INDEXLIST, Constants.NBT.TAG_INT);
+
+        if (handlerList.tagCount() == handlers.length){
+            for (int i = 0; i < handlerList.tagCount(); i++)
+            {
+                final NBTTagCompound handlerCompound = handlerList.getCompoundTagAt(i);
+                final IItemHandlerModifiable modifiable = handlers[indexList.getIntAt(i)];
+                if (modifiable instanceof INBTSerializable)
+                {
+                    INBTSerializable serializable = (INBTSerializable) modifiable;
+                    serializable.deserializeNBT(handlerCompound);
+                }
+            }
+        }
+
+        setName(nbt.hasKey(NBT_KEY_NAME) ? nbt.getString(NBT_KEY_NAME) : null);
+    }
+
+    /**
+     * Overrides the stack in the given slot. This method is used by the
+     * standard Forge helper methods and classes. It is not intended for
+     * general use by other mods, and the handler may throw an error if it
+     * is called unexpectedly.
+     *
+     * @param slot  Slot to modify
+     * @param stack ItemStack to set slot to (may be null)
+     * @throws RuntimeException if the handler is called in a way that the handler
+     *                          was not expecting.
+     **/
+    @Override
+    public void setStackInSlot(int slot, final ItemStack stack)
+    {
+        for(IItemHandlerModifiable modifiable : handlers) {
+            if(slot < modifiable.getSlots()) {
+                modifiable.setStackInSlot(slot, stack);
+                return;
+            }
+
+            slot -= modifiable.getSlots();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "slot". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1226)